internal/expr/parser.go

```diff
@@ -525,26 +475,26 @@ func (p *Parser) parseTargetType() (memberAccessor, bool, error) {
 }
 
 // parseSliceAccessor parses a slice range composed of the form "[:]".
-func (p *Parser) parseSliceAccessor() (sa sliceAccessor, ok bool, err error) {
+func (p *Parser) parseSliceAccessor() (string, bool, error) {
```

Comment:
Can we make this return `sliceAccessor`? I think it makes the method clearer, you dont have to state that the string you return is the type name which is then to be wrapped by `sliceAccessor`

Reply:
I've changed this to return a `sliceInputExpr` instead since that directly contains the slice type name as a string. Because of the name I've changed it to also parse the `$` rather than doing that in `parseInputExpr` (which also makes sense with the INSERT input expression being added soon)

Reply:
I will try and define what a `sliceAccessor` is better in the comment above (i.e. `Slice[:]` without a `$` or `&`), but for the reasons above, I think the function name makes more sense as it is.

Reply:
Have changed it to that.